Bindings/Java/tests/TestTables.java

```diff
@@ -303,12 +359,87 @@ public static void test_TimeSeriesTableVec3() {
+        assert tableDouble.getRowAtIndex(1).get( 5) == 2;
+        assert tableDouble.getRowAtIndex(1).get(11) == 2;
+        StdVectorString suffixes = new StdVectorString();
+        suffixes.add("_x"); suffixes.add("_y"); suffixes.add("_z");
+        tableDouble = table.flatten(suffixes);
+        assert tableDouble.getNumRows() == 2;
+        assert tableDouble.getNumColumns() == 12;
+        assert tableDouble.getColumnLabels().size() == 12;
+        assert tableDouble.getColumnLabel( 0).equals("0_x");
+        assert tableDouble.getColumnLabel( 1).equals("0_y");
+        assert tableDouble.getColumnLabel( 2).equals("0_z");
+        assert tableDouble.getColumnLabel( 3).equals("1_x");
+        assert tableDouble.getColumnLabel( 4).equals("1_y");
+        assert tableDouble.getColumnLabel( 5).equals("1_z");
+        assert tableDouble.getColumnLabel( 6).equals("2_x");
+        assert tableDouble.getColumnLabel( 7).equals("2_y");
+        assert tableDouble.getColumnLabel( 8).equals("2_z");
+        assert tableDouble.getColumnLabel( 9).equals("3_x");
+        assert tableDouble.getColumnLabel(10).equals("3_y");
+        assert tableDouble.getColumnLabel(11).equals("3_z");
+        assert tableDouble.getRowAtIndex(0).get( 0) == 1;
+        assert tableDouble.getRowAtIndex(0).get( 5) == 1;
+        assert tableDouble.getRowAtIndex(0).get(11) == 1;
+        assert tableDouble.getRowAtIndex(1).get( 0) == 2;
+        assert tableDouble.getRowAtIndex(1).get( 5) == 2;
+        assert tableDouble.getRowAtIndex(1).get(11) == 2;
     }
 
-    public static void main(String[] args) {
+    public static void test_FlattenWithIK() throws java.io.IOException {
+        String setupFileName = new String("subject01_Setup_IK_generic.xml");
+        String markerFileName = new String("walk_free_01.trc");
+        String modelFileName = new String("subject01_gait2392_scaled.osim");
+
+        TRCFileAdapter trcAdapter = new TRCFileAdapter();
+        TimeSeriesTableVec3 markerTable = trcAdapter.read(markerFileName);
+        StdVectorString suffixes = new StdVectorString();
+        suffixes.add(".x"); suffixes.add(".y"); suffixes.add(".z");
+        TimeSeriesTable markerTableFlat = markerTable.flatten(suffixes);
+        STOFileAdapter stoAdapter = new STOFileAdapter();
+        markerFileName = "walk_free_01.sto";
+        stoAdapter.write(markerTableFlat, markerFileName);
+
+        InverseKinematicsTool ikTool = new InverseKinematicsTool(setupFileName);
+        ikTool.setName("ik_test");
+        ikTool.setModel(new Model(modelFileName));
+        ikTool.setMarkerDataFileName(markerFileName);
+        StdVectorDouble timeColumn = markerTable.getIndependentColumn();
+        ikTool.setStartTime(timeColumn.get(0));
+        ikTool.setEndTime(timeColumn.get((int)(timeColumn.size() - 1)));
+        String ikResultsFileName = new String("ik_results.mot");
+        ikTool.setOutputMotionFileName(ikResultsFileName);
+
+        ikTool.run();
```

Comment:
I like having this here and it is reassuring that the test executes as expected. If there are other scripting tests that verify that the tool produces the expected results, then I don't think it is necessary to add it here. @jimmyDunne looks like you were able to successfully use the flattening ability to input forces into the tools, which is awesome. Can you indicate where that is being done (now that #1286 into this PR is closed)  for completeness.